### PR TITLE
CMP+CRMF: Re-formulate and (re-)add some comments removed by PR #15539

### PR DESCRIPTION
**Pull request by DDvO**
Those comments had been removed by #15539 because they had the word 'TODO' in them.
They still do have their merits, so I re-formulated them accordingly.

### CODE REVIEW

**Comment by t-j-h** [comment]
If you have a pile of things which you think should be improved in the code then raise github issues for tracking this.
Comments in the code tend to get completely ignored - and we work off github issues - that is how work gets completed.
Adding these things back into the code will not improve the change that they will be fixed - raising github issues will.

**Comment by mattcaswell** [comment]
OTC: Moving to this to Post 3.0 milestone

**Comment by DDvO** [comment]
Rebased to fix trivial merge conflict.
Ping for reviews. 
This just (re-)adds some comments and moves an aux function to `apps/lib/apps.c`.

**Comment by DDvO** [comment]
Carved out the code cleanup to #18900.
So this PR now just adds a bunch of comments.

**Review by t8m — DISMISSED** [review]

**Comment by t8m** [comment]
IMO only the comments which can hint the implementer of a future feature should be added. I.E., do not add comments that are really RFEs by themselves.

**Comment by DDvO** [comment]
> IMO only the comments which can hint the implementer of a future feature should be added. I.E., do not add comments that are really RFEs by themselves.

Thanks @t8m for having a look again and form your comments - 
when I have some spare time, I will convert the code comments that hint on feature extensions elsewhere, such as proper FRs.

**Comment by DDvO** [comment]
Rebased to fix merge conflict.
Converted one comment to a question: #19805 and another one to a FR: #19087.
Removed, re-formulated, or solved the rest.

**Comment by DDvO** [comment]
Please review - it's a pretty low-hanging fruit.

File: crypto/crmf/crmf_local.h [hunk 1]
```diff
@@ -329,7 +330,7 @@ struct ossl_crmf_certtemplate_st {
 struct ossl_crmf_certrequest_st {
     ASN1_INTEGER *certReqId;
     OSSL_CRMF_CERTTEMPLATE *certTemplate;
-    STACK_OF(OSSL_CRMF_ATTRIBUTETYPEANDVALUE) *controls;
+    STACK_OF(OSSL_CRMF_ATTRIBUTETYPEANDVALUE /* Controls expanded */) *controls;
```
Comment (beldmit) [inline crypto/crmf/crmf_local.h:333]: Not sure about style here.

**Review by beldmit — APPROVED** [review]
LGTM, have one minor style question

File: crypto/crmf/crmf_local.h (hunk 1 above)
Comment (DDvO) [reply crypto/crmf/crmf_local.h:333]: At least `check-format.pl crypto/crmf/crmf_local.h` is happy with it.

**Review by hlandau — APPROVED** [review]

**Comment by DDvO** [comment]
Merged -  thanks @beldmit and @hlandau.

**Pull request closed without merging**